lang/py/avro/schema.py

```diff
@@ -158,7 +159,77 @@ def canonical_properties(self):
             if key in props)
 
 
-class Schema(abc.ABC, CanonicalPropertiesMixin):
+class FingerprintMixin:
+    """
+    A Mixin to generate schema fingerprints for supported algorithms
+    """
+    _FP_TABLE = None
+    _EMPTY64 = 0xc15d213aa4d7a795
+
+    # All algorithms guaranteed by hashlib are supported
+    SUPPORTED_ALGORITHMS = set(hashlib.algorithms_guaranteed)
```

Comment:
The list of guaranteed algorithms is:
```
 'blake2b', 
 'blake2s',
 'md5',
 'sha1',
 'sha224',
 'sha256',
 'sha384',
 'sha3_224',
 'sha3_256',
 'sha3_384',
 'sha3_512',
 'sha512',
 'shake_128',
 'shake_256'
```

- Should all of these be allowed, or should there be a restricted list? 
- Algorithm names are treated case-insensitively. Are there algorithm name variations to be supported for compatibility with other language implementations?

Reply:
1. I think they should all be allowed.
2. This value is already a set. If you want to ensure it can't be mutated, better change it to a `frozenset`.
3. _Are there algorithm name variations to be supported for compatibility with other language implementations?_ I don't know.